packages/ibm-products/src/components/Datagrid/useDisableSelectRows.ts

```diff
@@ -0,0 +1,119 @@
+/*
+ * Licensed Materials - Property of IBM
+ * 5724-Q36
+ * (c) Copyright IBM Corp. 2020
+ * US Government Users Restricted Rights - Use, duplication or disclosure
+ * restricted by GSA ADP Schedule Contract with IBM Corp.
+ */
+
+import { Hooks, Row, RowPropGetter, TableRowProps } from 'react-table';
+import { DataGridTableInstance, DatagridRow } from './types';
+
+interface PropGetterMeta {
+  instance?: DataGridTableInstance;
+  row?: Partial<Row<any> & DatagridRow<any>>;
+}
+
+const useDisableSelectRows = (hooks: Hooks) => {
+  updateSelectAll(hooks);
+  updatePageSelectAll(hooks);
+
+  const getRowProps: RowPropGetter<any> = (
+    props: Partial<TableRowProps>,
+    { row, instance }: PropGetterMeta
+  ) =>
+    [
+      props,
+      {
+        disabled: instance?.shouldDisableSelectRow?.(row),
+      },
+    ] as Partial<TableRowProps>[];
+
+  hooks.getRowProps.push(getRowProps);
+};
+
+const updateSelectAll = (hooks: Hooks) => {
+  const getToggleAllRowsSelectedProps: RowPropGetter<any> = (
+    props: Partial<TableRowProps>,
+    { instance }: PropGetterMeta
+  ) => {
+    const selectableRows =
```

Comment:
type not declared for `selectableRows`

Reply:
It is inferring the type `Row<any>`, so there is no need to explicitly provide a type.